crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -88,6 +90,18 @@ public struct DenyCapV2<phantom T> has key, store {
     allow_global_pause: bool,
 }
 
+/// A dynamic field key added to `CoinMetadata` objects created after protocol version 63.
+/// This is bound to a `RegulatedInfo` enum
+public struct RegulatedMarker has copy, drop, store {}
+
+/// Auxiliary metadata about the regulated operations supported by the coin (if any)
+public enum RegulatedInfo has copy, drop, store {
+    /// An ordinary coin that does not support a denylist or global pause
+    Ordinary,
+    /// A regulated coin that supports a denylist, and (maybe) global paus
+    Regulated { allow_global_pause: bool}
```

Comment:
What if we did something like 
```suggestion
    Regulated { deny_cap: ID }
```
which points to the `DenyCapV2`. Obviously some issues if the object is wrapped/deleted, but the flag `allow_global_pause` cannot be modified after creation, so it is fine to look at the last time the `DenyCap` existed. 

My goal here is to allow existing coins (at least regulated ones) to add their `RegulatedInfo` after the fact.

Reply:
The intention of this change is to allow Move code to special-case regulated coins (e.g., DEX pool creation, which typically already asks for `CoinMetadata` to get the `decimals`). If the Move code wants to treat global pause coins differently, it needs a boolean, not a pointer.

Would be fine adding the deny cap ID as additional metadata--more info seems better for sure